src/java.base/share/classes/java/util/zip/ZipCoder.java

```diff
@@ -158,7 +158,7 @@ static int hash(String name) {
         return hsh;
     }
 
-    boolean hasTrailingSlash(byte[] a, int end) {
+    protected boolean hasTrailingSlash(byte[] a, int end) {
```

Comment:
Why are you making these `protected`? `ZipCoder` is package-private so any inheritors must be in the same package, which means they already have access to package-private methods.

Reply:
Since `hasTrailingSlash` is now only used from UTFCoder and subclasses, I thought we could stricten the access to protected. But as I learned, `protected` is still accessible from the same package.

On closer inspection though, `hasTrailingSlash` is now only used from UTF8ZipEncoder.compare. So we can actually make that implementation `private` and remove the now-unused implementation from the base class, along with the `slashBytes` logic. I have pushed these changes.

What do you think?